cmd/bosun/main.go

```diff
@@ -137,19 +143,69 @@ func main() {
 			}))
 			slog.Infoln("readonly relay at", ts.URL, "to", tsdbHost)
 			tsdbHost, _ = url.Parse(ts.URL)
-			c.TSDBHost = tsdbHost.Host
+			sysProvider.SetTSDBHost(tsdbHost.Host)
 		}
 	}
-	if c.InternetProxy != "" {
-		web.InternetProxy, err = url.Parse(c.InternetProxy)
+	if sysProvider.GetInternetProxy() != "" {
+		web.InternetProxy, err = url.Parse(sysProvider.GetInternetProxy())
 		if err != nil {
 			slog.Fatalf("InternetProxy error: %s", err)
 		}
 	}
-	if *flagQuiet {
-		c.Quiet = true
+	cmdHook, err := conf.MakeSaveCommandHook(sysProvider.GetCommandHookPath())
+	if err != nil {
+		slog.Fatal(err)
+	}
+	ruleProvider.SetSaveHook(cmdHook)
```

Comment:
Why does the ruleProvider need to be given the save hook. Can't the reload function below just execute it?